### ISSUE TITLE
EncryptionRequestHandler support for collection with multiple shards

### ISSUE THREAD

**Issue by danielsason112** [apache/solr-sandbox#110]
Hi, running Solr cloud mode, and created a collection with multiple shards. When calling the EncryptionRequestHandler, the encryption process is invoked only for the core that handles the request.

What is the right approach for handling a multi shards collection?
Call /admin/encrypt API for each of the cores separately?

Appreciate suggestions and help on this one. Thanks.

**Comment by dsmiley** [comment]
Correct, invoke on each. This could be improved. 
cc @bruno-roustant

**Comment by bruno-roustant** [comment]
Yes, calling it on each leader core separately is the right way to do it currently.
There is a plan to improve that, but not yet available.

**Comment by bruno-roustant** [comment]
I created a [PR](https://github.com/apache/solr-sandbox/pull/115) to encrypt a collection with multiple shards in a single request.
